loader/loader.go

```diff
@@ -127,12 +127,29 @@ func New(dir string, paths ...string) (*Loader, error) {
 //
 // Generated files are written to the same directory, next to the source gunk
 // files.
+//
+// It is fine to pass every 'plugin.CodeGeneratorRequest' to every protoc
+// generator unaltered; this is what protoc does when calling out to the
+// generators and the generators should already handle the case where
+// they aren't required.
 func (l *Loader) GeneratePkg(path string) error {
 	req := l.requestForPkg(path)
-	bs, err := proto.Marshal(req)
+	if err := l.generatePluginGo(*req); err != nil {
+		return fmt.Errorf("error generating plugin go: %v", err)
+	}
+	if err := l.generatePluginGrpcGateway(*req); err != nil {
+		return fmt.Errorf("error generating plugin grpc-gateway: %v", err)
+	}
+	return nil
+}
+
+func (l *Loader) generatePluginGo(req plugin.CodeGeneratorRequest) error {
+	req.Parameter = proto.String("plugins=grpc")
```

Comment:
can we not leave this in for the request in all generators?

Reply:
Not for `grpc-gateway` at least, it fails horribly.